ibm/resource_ibm_cis_origin_pool.go

```diff
@@ -2,300 +2,383 @@ package ibm
+	cisGLBPoolHealth               = "health"
+	cisGLBPoolHealthy              = "healthy"
+	cisGLBPoolCreatedOn            = "created_on"
+	cisGLBPoolModifiedOn           = "modified_on"
+	cisGLBPoolOriginsName          = "name"
+	cisGLBPoolOriginsAddress       = "address"
+	cisGLBPoolOriginsEnabled       = "enabled"
+	cisGLBPoolOriginsHealthy       = "healthy"
+	cisGLBPoolOriginsWeight        = "weight"
+	cisGLBPoolOriginsDisabledAt    = "disabled_at"
+	cisGLBPoolOriginsFailureReason = "failure_reason"
 )
 
 func resourceIBMCISPool() *schema.Resource {
 	return &schema.Resource{
 		Schema: map[string]*schema.Schema{
-			"cis_id": {
+			cisID: {
 				Type:        schema.TypeString,
 				Description: "CIS instance crn",
 				Required:    true,
 			},
-			"name": {
+			cisGLBPoolID: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			cisGLBPoolName: {
 				Type:        schema.TypeString,
 				Description: "name",
 				Required:    true,
-				//ValidateFunc: validation.StringLenBetween(1, 32),
 			},
-			"check_regions": {
+			cisGLBPoolRegions: {
 				Type:     schema.TypeSet,
 				Required: true,
 				Elem: &schema.Schema{
 					Type: schema.TypeString,
 				},
 				Description: "List of regions",
 			},
-			"description": {
+			cisGLBPoolDesc: {
 				Type:        schema.TypeString,
 				Optional:    true,
 				Description: "Description of the CIS Origin Pool",
 			},
-			"enabled": {
+			cisGLBPoolEnabled: {
 				Type:        schema.TypeBool,
 				Required:    true,
 				Description: "Boolean value set to true if cis origin pool needs to be enabled",
 			},
-			"minimum_origins": {
+			cisGLBPoolMinimumOrigins: {
 				Type:        schema.TypeInt,
 				Optional:    true,
 				Default:     1,
 				Description: "Minimum number of Origins",
 			},
-			"monitor": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Description: "Monitor value",
+			cisGLBPoolMonitor: {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Description:      "Monitor value",
+				DiffSuppressFunc: suppressDomainIDDiff,
 			},
-			"notification_email": {
+			cisGLBPoolNotificationEMail: {
 				Type:        schema.TypeString,
 				Optional:    true,
 				Description: "Email address configured to recieve the notifications",
 			},
-			"origins": {
+			cisGLBPoolOrigins: {
 				Type:        schema.TypeSet,
 				Required:    true,
 				Description: "Origins info",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"name": {
+						cisGLBPoolOriginsName: {
 							Type:     schema.TypeString,
 							Required: true,
 						},
-						"address": {
+						cisGLBPoolOriginsAddress: {
 							Type:     schema.TypeString,
 							Required: true,
 						},
-						"enabled": {
+						cisGLBPoolOriginsEnabled: {
 							Type:     schema.TypeBool,
 							Required: true,
 						},
-						"weight": {
-							Type:     schema.TypeInt,
+						cisGLBPoolOriginsWeight: {
+							Type:     schema.TypeFloat,
```

Comment:
Why is it changed to float any specific reason?

Reply:
weight valid values are between `0.0` to `1.0`.